### DISCUSSION TITLE
Unable to import frozen modules even though they show up in help('modules')

### DISCUSSION THREAD

**Discussion by alvarop** [micropython/micropython#11732] [Core Development] [answered]
I'm working on porting micropython v1.20.0 on a pre-existing FreeRTOS based project. I based it off the minimal port. It's running on an STM32U575 (Cortex-M33)

I have the following working:
* REPL
  *  I can run a repl, call functions, and import some modules
* c modules
  * I have a custom c module (outside the micropython path, but linked as a library) working and I can import it
* _some_ of the micropython-lib things seem to be working, but I'm not entirely sure

Once I started trying to do some mpy-cross and .mpy files, I started running into issues.

I tried to create a frozen manifest.py with something like:
```py
module("testmod.py", base_path="modules/testmod/")
require("base64")
``` 

The project compiles ok and I can see them listed when I run `help('modules')`, but if I try to import them, I get a "no module named" error.

![image](https://github.com/micropython/micropython/assets/744129/71152efd-87a6-44e7-b263-0ee7d5903ee8)

In the image above, base64 and testmod are frozen modules. Base64 should come from micropython-lib and testmod is just a simple python file inside the `modules/testmod/` directory and looks something like:

```py

class TestMod:
    def __init__(self):
        self.test = 1

    def print(self):
        print("HELLO")
```

*device* is my custom C module which is working fine.

Any idea where I could be going wrong?

Happy to share more details of the build, but it's a fairly complicated cmake project so I don't want to provide _too much_ information if it's something simple I'm missing 😅

Also, if it's better _not_ do pre-compile and there's a better way to import them, I'm all ears! I don't yet have the filesystem working, so it's all pre-compiled fun right now.

**Comment by jimmo** [comment, marked as answer]
@alvarop Can you inspect the value of `sys.path`. My guess is that you don't have `".frozen"` in it.

**Reply by alvarop** [reply]
Oh wow, I didn't have sys! For some reason I had disabled it earlier with `#define MICROPY_PY_SYS 0` and never re-enabled it.
Setting it back to 1 fixed the problem. Thank you so much!

**Reply by jimmo** [reply]
OK great!

> I based it off the minimal port.

@alvarop FYI btw have you seen the new `embed` port? It's still a WIP but it's designed for the purpose of embedding MicroPython into existing RTOSes and build systems.

**Reply by alvarop** [reply]
I have! I started with the minimal, but when I saw v1.20.0 was adding the embed, I waited a bit and used that as reference too :)
